addons/mail/static/src/js/text_emojis_fields.js

```diff
@@ -0,0 +1,138 @@
+odoo.define('mail.text_emojis_field', function (require) {
+"use strict";
+
+var basicFields = require('web.basic_fields');
+var core = require('web.core');
+var emojis = require('mail.emojis');
+var registry = require('web.field_registry');
+var MailEmojisMixin = require('mail.emoji_mixin');
+var _onEmojiClickMixin = MailEmojisMixin._onEmojiClick;
+var QWeb = core.qweb;
+
+/*
+ * Common code for FieldText and FieldChar
+ */
+var FieldEmojiCommon = {
+    /**
+     * @override
+     * @private
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        this._updatePreview =_.throttle(this._updatePreview, 1000, {leading: false});
+        this.emojis = emojis;
+    },
+
+    /**
+     * This will add an emoji button that shows the emojis selection dropdown.
+     *
+     * We use 'on_attach_callback' because we need the element to be attached to the form first.
+     * That's because the $emojisIcon element needs to be rendered outside of this $el
+     * (which is an text element, that can't 'contain' any other elements).
+     *
+     * @override
+     */
+    on_attach_callback: function () {
+        var self = this;
+        if (!this.$emojisIcon) {
+            this.$emojisIcon = $(QWeb.render('mail.EmojisDropdown', {widget: this}));
+            this.$emojisIcon.find('.o_mail_emoji').on('click', function (ev) {
+                self._onEmojiClick(ev);
+                self._isDirty = true;
```

Comment:
The `_isDirty` state could be set by the `_onEmojiClick` function.

So you can change to `this.$emojisIcon.find('.o_mail_emoji').on('click', this._onEmojiClick.bind(this));` which is cleaner.